File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredIntegerPropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KRequiredIntegerPropertyName implements KRequiredPropertyName<Integer>, KSortablePropertyName<Integer> {

    private final String name;

    public KRequiredIntegerPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredIntegerPropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KRequiredIntegerPropertyName(val name: String) : KRequiredPropertyName<Int>, KSortablePropertyName<Int> {

    override fun name(): String = name

}
